lib/Serialization/Deserialization.cpp

```diff
@@ -4148,7 +4163,21 @@ Expected<Type> ModuleFile::getTypeChecked(TypeID TID) {
     decls_block::BoundGenericTypeLayout::readRecord(scratch, declID, parentID,
                                                     rawArgumentIDs);
 
-    auto nominal = cast<NominalTypeDecl>(getDecl(declID));
+    auto *decl = getDecl(declID);
+    if (auto *alias = dyn_cast<TypeAliasDecl>(decl)) {
+      // FIXME: We should substitute through the alias rather than just relying
+      // on it forwarding arguments perfectly to the nominal.
+      auto *nominal = alias->getUnderlyingTypeLoc().getType()->getAnyNominal();
```

Comment:
I think `alias->getDeclaredInterfaceType()->getAnyNominal()` is better. It should be equivalent.